frontend/src/pages/MainPage/components/Banner/Banner.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useRef, useState, useEffect } from 'react';
+import * as Styled from './Banner.styles';
+import { BannerProps } from './Banner.styles';
+
+interface BannerComponentProps {
+  banners: BannerProps[];
+}
+
+const Banner = ({ banners }: BannerComponentProps) => {
+  const slideRef = useRef<HTMLDivElement>(null);
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const IMG_WIDTH = 1180;
+
+  useEffect(() => {
+    if (slideRef.current) {
+      slideRef.current.style.transform = `translateX(-${currentSlide * IMG_WIDTH}px)`;
+    }
+  }, [currentSlide]);
+
+  const moveToNextSlide = () => {
+    if (currentSlide >= banners.length - 1) return;
+    setCurrentSlide((prev) => prev + 1);
+  };
+
+  const moveToPrevSlide = () => {
+    if (currentSlide <= 0) return;
+    setCurrentSlide((prev) => prev - 1);
+  };
```

Comment:
전 후 버튼으로 슬라이드 전환 로직이 깔끔하게 잘 작성되어 있네요!

추가로, 자료구조 원형 큐 방식처럼 % 연산을 활용하면 마지막 슬라이드에서 자동으로 처음으로 넘어가는 형태의 슬라이드로 만들 수 있어여
```suggestion
const moveToNextSlide = () => {
  setCurrentSlide((prev) => (prev + 1) % banners.length);
};

const moveToPrevSlide = () => {
  setCurrentSlide((prev) => (prev - 1 + banners.length) % banners.length);
};
```

Reply:
오호 바로 적용해보겠습니다